xrpl/asyncio/transaction/multisign.py

```diff
@@ -0,0 +1,73 @@
+"""Multisign transaction methods with XRPL transactions."""
+from typing import List
+
+from xrpl.core.binarycodec import decode, encode, encode_for_multisigning
+from xrpl.core.keypairs.main import sign
+from xrpl.models.transactions.transaction import Signer, Transaction
+from xrpl.wallet.main import Wallet
+
+
+async def sign_for_multisign(
+    transaction: Transaction,
+    wallet: Wallet,
+) -> str:
+    """
+    Signs a transaction to be used for multisigning.
+
+    Args:
+        transaction: the transaction to be signed.
+        wallet: the wallet with which to sign the transaction.
+
+    Returns:
+        The signed transaction blob.
+    """
+    signature = sign(
+        bytes.fromhex(
+            encode_for_multisigning(
+                transaction.to_xrpl(),
+                wallet.classic_address,
+            )
+        ),
+        wallet.private_key,
+    )
+
+    tx_dict = transaction.to_dict()
+    tx_dict["signers"] = [
+        Signer(
+            account=wallet.classic_address,
+            txn_signature=signature,
+            signing_pub_key=wallet.public_key,
+        )
+    ]
+
+    return encode(Transaction.from_dict(tx_dict).to_xrpl())
+
+
+async def multisign(transaction: Transaction, tx_blobs: List[str]) -> Transaction:
```

Comment:
Same question here - does this function need to be async?

Reply:
If the function doesn't require async methods, then should it be removed from asyncio folder entirely? So if someone was using Async client, they would just use sync version of it?

Reply:
That seems reasonable